libfwupdplugin/fu-archive.c

```diff
@@ -234,8 +257,136 @@ fu_archive_load(FuArchive *self, GBytes *blob, FuArchiveFlags flags, GError **er
 #endif
 }
 
+static GByteArray *
+fu_archive_save(FuArchive *self,
+		const gchar *compress,
+		FuArchiveEntryIterateFunc callback,
+		gpointer user_data,
+		GError **error)
+{
+#ifdef HAVE_LIBARCHIVE
+	const gchar *fn;
+	gsize blobsz;
+	size_t size;
+	int r;
+	g_autoptr(GByteArray) blob = NULL;
+	g_autoptr(_archive_write_ctx) arch = NULL;
+
+	blobsz = 0x20000;
+	blob = g_byte_array_sized_new(blobsz);
```

Comment:
Find a way to resize buffer with libarchive.

Reply:
I can do this if you want.

Reply:
okay